docs: GraphQL Fragments recipe (#16659)
gatsbyjs/gatsby: docs/docs/recipes.md

- [Gatsby GraphQL reference for aliasing](/docs/graphql-reference/#aliasing)
- Live example:

<iframe
  title="Using aliases"
  src="https://711808k40x.sse.codesandbox.io/___graphql?query=%7B%0A%20%20fileCount%3A%20allFile%20%7B%20%0A%20%20%20%20totalCount%0A%20%20%7D%0A%20%20filePageInfo%3A%20allFile%20%7B%0A%20%20%20%20pageInfo%20%7B%0A%20%20%20%20%20%20currentPage%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A&explorerIsOpen=false"
  width="600"
  height="300"
/>

### GraphQL Query Fragments

GraphQL fragments are shareable chunks of a query that can be reused.

> You might want to use them to share multiple fields between queries or to co-locate the data with the component it's used in.

Reviewer: ```suggestion
You might want to use them to share multiple fields between queries or to colocate a component with the data it uses. 
```

#### Directions

> 1. Declare a `graphql` template string with a Fragment in it. The fragment should be made up of the keyword fragment, a name, and the GraphQL type it is associated with (in this case of type `Site`, demonstrated by `on Site`), and the fields that make up the fragment inside:

Reviewer: ```suggestion
1. Declare a `graphql` template string with a Fragment in it. The fragment should be made up of the keyword `fragment`, a name, the GraphQL type it is associated with (in this case of type `Site`, as demonstrated by `on Site`), and the fields that make up the fragment:
```

```jsx
export const query = graphql`
  // highlight-start
  fragment SiteInformation on Site {
    title
    description
  }
  // highlight-end
`
```

> 2. Then you can include the fragment in a query for a field of the type specified by the fragment, to include those fields without you having to type them all out:

Reviewer: ```suggestion
2. Now, include the fragment in a query for a field of the type specified by the fragment. This includes those fields without having to declare them all independently:
```

```diff
export const pageQuery = graphql`
  query SiteQuery {
    site {
-     title
-     description
+   ...SiteInformation
    }
  }
`
```

> **Note**: if the fragment is used in a different file it doesn't need to be imported. Exporting the query from the original file will make it available for queries across your project.

Reviewer: ```suggestion
**Note**: Fragments don't need to be imported in Gatsby. Exporting a query with a Fragment makes that Fragment available in _all_ queries in your project.
```
